src/main/java/com/google/devtools/build/lib/rules/cpp/CcLibrary.java

```diff
@@ -615,7 +618,7 @@ public static boolean appearsToHaveObjectFiles(AttributeMap rule) {
               ccToolchain,
               configuration,
               Link.LinkTargetType.STATIC_LIBRARY,
-              /* linkedArtifactNameSuffix= */ ""));
+              linkedArtifactNameSuffix));
```

Comment:
Ah... we should not add suffix to static library. But I looked at the code and found no good way to only apply   linkedArtifactNameSuffix to dynamic library in CcLinkingHelper. Looks like we still have to have an additional attribute for this. They we can apply the DLL suffix based on the `linkTargetType`. Sorry about the back and forth.